code/modules/projectiles/updated_projectiles/gun_system.dm

```diff
@@ -217,16 +222,17 @@
 	if(!(flags_item & TWOHANDED) || flags_item & WIELDED)
 		return
 
-	var/obj/item/offhand = user.get_inactive_hand()
-	if(offhand)
-		if(offhand == user.r_hand)
-			user.drop_r_hand()
-		else if(offhand == user.l_hand)
-			user.drop_l_hand()
-		if(user.get_inactive_hand()) //Failsafe; if there's somehow still something in the off-hand (undroppable), bail.
-			to_chat(user, "<span class='warning'>You need your other hand to be empty!</span>")
-			return
-			
+	if(user.get_inactive_hand())//Drop item held in the off-hand if occupied.
+		var/obj/item/offhand = user.get_inactive_hand()
```

Comment:
Not to call the same proc twice you can define the var first:
`var/obj/item/offhand`
And then run the control statement: `if(offhand = user.get_inactive_hand())`
It will check the proc, give the value to `offhand` and proceed if not null or false.